tools/buildHelper/buildHelper.psm1

```diff
@@ -775,6 +758,68 @@ function Get-TagData
     }
 }
 
+function Invoke-FormatTagTemplate {
+    param(
+        [parameter(Mandatory)]
+        [string]
+        $TagTemplate,
+
+        [parameter(Mandatory)]
+        [object]
+        $TagGroup,
+
+        [parameter(Mandatory)]
+        [string]
+        $Version
+    )
+
+    $currentTag = $TagTemplate
+
+    $fullTags = $tagGroup.Group | Where-Object {$_.Type -eq 'Full'}
+
+    $actualMatrixTagTemplates = @()
+    foreach($tag in $tagGroup.Group) {
+        # replace the tag token with the tag
+        if ($currentTag -match '#tag#') {
+            $actualMatrixTagTemplates += $currentTag -replace '#tag#', $tag.Tag
```

Comment:
while I don't usually prefer the api approach, it might be better to do:
`$currentTag.Replace("#tag#",$tag.Tag)`
It's less forgiving, which may be what you want in this circumstance.